text/0000-afp.md

```diff
@@ -0,0 +1,354 @@
+
+The requester would first find a set of farmers who have the correct software for transcoding a video. The requester would then pass those farmers to their AFP implementation. AFP would then ask each peer their transcode cost per frame. The matcher implementation could then find a set of peers whose total cost is less than a desired amount and then hire that set of farmers. The AFP implementation could then take subset of farmers who have contractually agreed to do work for the requester and start the video transcode.
+
+## 6. Drawbacks
+
+There are many methods for determining cost in an open market, so any limitation on how cost is determined may result in a direct or indirect impact on the market itself. In this RFC, on a per-contract basis, the requester asks for a single non-updateable bid for a statement-of-work from each farmer. This gives an advantage to the requester because it allows them to choose a lower cost farmer (even if the requester was initially willing to pay more). It also does not allow the farmer to adjust their bid in real-time for the current contract, nor does it allow the farmer to compare their bid with the bids of other farmers for the current contract. The farmer, however, can adjust their bid prior to the next work request, and through AFP, can check to see the cost of nearby peers to stay competitive. This means that this cost determination method should have no significant impact on the larger market. 
+
+## 7. Alternatives
+
+An alternative to a single bid system would be to have a third party be the moderator of a bid, such as is used in real-time bidding with advertising. 
+
+Another alternative would be to have a public marketplace where farmers and requesters list their service requests/offers, and use this marketplace to find workers/work.
+
+## 8. Adoption Strategy
+
+Ara Farming Protocol is hosted on Github, under AraBlocks organization, and distributed as an open source project along with other Ara core components. AFP is not a required component for using Ara, but it is currently being extended with DCDN to allow for rewardable file distribution. 
+
+## 9. Unresolved Questions
+
+### 9.1. How would contracts without defined ends (such as file storage) use AFP?
+
+These sort of contracts could be set to deliver rewards in discretized intervals (such as a monthly basis). The service could be setup to renegotiate contracts at these intervals based on current market costs. Guaranteeing uninterrupted service would be the responsibility of the developer of the service. 
+
+### 9.2. How would AFP deter farmers/requesters from gaming the system?
+
+If it is possible for farmers to request the bids of other farmers through AFP, then it is possible that farmers could simply underbid their neighbors to guarantee that they receive the work. While this is possible, it is really a question of the market (scarcity of work vs. scarcity of farmers) if this will actually be problem.  
+
+## 10 Security Considerations
```

Comment:
Maybe a general note siting that each specific implementation has its own security considerations (duplex streams, gRPC, etc).